### PR TITLE
Overloading of some method to use RNGCryptoServiceProvider class

### PR DESCRIPTION
**Pull request by cheeyixu**
Edit by @bazzilic: this PR is in support of issue https://github.com/bazzilic/PaillierExt/issues/3

### CODE REVIEW

File: BigInteger/BigInteger.cs [hunk 1]
```diff
@@ -1652,6 +1653,37 @@ public void genRandomBits(int bits, Random rand)
 
 
     /// <summary>
+    /// Populates "this" with the specified amount of random bits (secured version)
+    /// </summary>
+    /// <param name="bits"></param>
+    /// <param name="rand"></param>
```
Comment (bazzilic) [inline BigInteger/BigInteger.cs]: Should be `rng`

**Comment by bazzilic** [comment]
These new methods are hard to test automatically. Did you do some manual tests to ensure that the methods behave as they are expected to?

**Comment by cheeyixu** [comment]
i did test the boundary values of the input, getRandomBits gives a huge integer indeed. but it is not behaving correctly when the input bit number is small. I am fixing it
